### PR TITLE
Project-mongo-API by Johan Löfgren & Katharina Liebig

### PR DESCRIPTION
**Pull request by katliebig**
One thing we were unsure about is error handling. Where exactly should we put try and catch?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 node_modules
 .DS_Store
+.env
```
Comment (newmarkdev) [inline .gitignore:3]: Well done on including both these in the gitignore 👍

File: server.js [hunk 2]
```diff
@@ -1,38 +1,170 @@
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import dotenv from "dotenv"
```
Comment (newmarkdev) [inline server.js:4]: not sure if this could be removed after connecting to heroku or if one or the other is better ?

**Review by newmarkdev — APPROVED** [review]
Well done, very cool project and data set ! 🎉😀  Dont have that much to add.

File: server.js (hunk 2 above)
Comment (KarolCislowski) [reply server.js:4]: dotenv package is injecting your process variables when you work on localhost.

File: server.js [hunk 3]
```diff
@@ -1,38 +1,170 @@
...
+app.get('/players', async (req, res) => {
+  try {
+    const players = await Player.aggregate(
+      [
+        // Filters for players with a ranking that is less than or equal as the query or 200 as default.
+        { $match: { ranking: { $lte: req.query.ranking ? +req.query.ranking : 200 } } },
+        { $group: { _id: "$name", doc: { $first: "$$ROOT" } } },
+        { $replaceRoot: { newRoot: "$doc" } },
+        { $sort: { ranking: 1 } }
+      ]
```
Comment (KarolCislowski) [inline server.js:71]: 😻

File: server.js [hunk 4]
```diff
@@ -1,38 +1,170 @@
...
+    )
+    res.json(players)
+  } catch {
+    res.status(400).json({ error: 'Something went wrong' })
+  }
+})
+
+// Endpoint for players by id
+app.get('/players/player/:id', async (req, res) => {
```
Comment (KarolCislowski) [inline server.js:80]: `/players/:id` would be more intuitive.

File: server.js [hunk 5]
```diff
@@ -1,38 +1,170 @@
...
+  try {
+    const countries = await Player.distinct('country')
+    res.json(countries)
+  } catch {
+    res.status(400).json({ error: 'Something went wrong' })
+  }
+})
+
+// Endpoint for country by country name
+app.get('/countries/:country', async (req, res) => {
```
Comment (KarolCislowski) [inline server.js:104]: you are showing the list of players from certain country so your endpoint should be:
`/players/country/:country`

File: server.js [hunk 6]
```diff
@@ -1,38 +1,170 @@
...
+              $and: [
+                { country: { $regex: new RegExp(req.params.country, "i") } },
+                { ranking: { $lte: req.query.ranking ? +req.query.ranking : 200 } }
+              ]
+            }
+          },
+          { $group: { _id: "$name", doc: { $first: "$$ROOT" } } },
+          { $replaceRoot: { newRoot: "$doc" } },
+          { $sort: { ranking: 1 } }
+        ]
```
Comment (KarolCislowski) [inline server.js:124]: 😻

File: server.js [hunk 7]
```diff
@@ -1,38 +1,170 @@
...
+      if (country.length > 0) {
+        res.json(country)
+      } else {
+        res.status(404).json({ error: 'No player within this ranking' })
+      }
+    } else {
+      res.status(404).json({ error: 'Country not found' })
+    }
+  } catch {
+    res.status(400).json({ error: 'Something went wrong' })
```
Comment (KarolCislowski) [inline server.js:135]: great choice of status codes 🥇

**Review by KarolCislowski — APPROVED** [review]
Great work with aggregated data 🥇

**Pull request closed without merging**